downstairs/src/lib.rs

```diff
@@ -1898,11 +2022,14 @@ impl Downstairs {
                     data,
                 }))
             }
-            IOop::WriteUnwritten { writes, .. } => {
+            IOop::WriteUnwritten { .. } => {
                 /*
                  * Any error from an IO should be intercepted here and passed
                  * back to the upstairs.
                  */
+                let IOop::WriteUnwritten { writes, .. } = job.work else {
```

Comment:
What's the purpose here of this check?  Are we not already matching on `job.work`?

Reply:
Congrats to all three reviewers for noticing this 😂 

https://github.com/oxidecomputer/crucible/pull/1272#discussion_r1600406065 explains the issue